src/EditorFeatures/CSharpTest/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -0,0 +1,748 @@
+{
+    public class MakeFieldReadonlyTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new CSharpMakeFieldReadonlyDiagnosticAnalyzer(), new CSharpMakeFieldReadonlyCodeFixProvider());
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        [InlineData("public")]
+        [InlineData("internal")]
+        [InlineData("protected")]
+        [InlineData("protected internal")]
+        public async Task NonPrivateField(string accessibility)
+        {
+            await TestMissingInRegularAndScriptAsync(
+$@"class MyClass
+{{
+    {accessibility} int[| _goo |];
+}}");
+        }
+        
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsEvent()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private event System.EventHandler [|Goo|];
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsReadonly()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private readonly int [|_goo|];
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsConst()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class MyClass
+{
+    private const int [|_goo|];
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldNotAssigned()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|];
+}",
+@"class MyClass
+{
+    private readonly int _goo;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldNotAssigned_Struct()
+        {
+            await TestInRegularAndScriptAsync(
+@"struct MyStruct
+{
+    private int [|_goo|];
+}",
+@"struct MyStruct
+{
+    private readonly int _goo;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldAssignedInline()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|] = 0;
+}",
+@"class MyClass
+{
+    private readonly int _goo = 0;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task MultipleFieldsAssignedInline_AllCanBeReadonly()
+        {
+            await TestInRegularAndScriptAsync(
+@"class MyClass
+{
+    private int [|_goo|] = 0, _bar = 0;
+}",
+@"class MyClass
+{
+    private readonly int _goo = 0;
+    private int _bar = 0;
```

Comment:
this seems unfortunate.  my preference would be to have at least two code actions.  one which splits things (and add 'readonly') to the one item.  The other which adds readonly to the whole declaration if applicable.  i.e. if all hte fields can become readonly, that should be something you can do without having to split hte declaration.

Reply:
Note: this is distinct from having "fix all".  Fix-all will scope to a file/project/solution.  i think it would be worthwhile to have a code action to just fix the fields in a single declaration (if possible).

Reply:
:memo: I believe this one can be a follow-up change.